packages/validation/src/validators/yaml-syntax/rules.ts

```diff
@@ -0,0 +1,311 @@
+import { ErrorCode } from "yaml";
+import { ValidationRule } from "../../common/sarif";
+import { NO_HELP_AVAILABLE } from "../../constants";
+
+/**
+ * Maps a YAML error code to its rule identifier.
+ *
+ * This is done because rule identifiers should be opaque.
+ * Let's use a consistent format so printing looks nice across validators.
+ *
+ * @see https://eemeli.org/yaml/#errors
+ */
+export const YAML_RULE_MAP: Record<ErrorCode | string, string> = {
+  ALIAS_PROPS: "YML001",
+  BAD_ALIAS: "YML002",
+  BAD_DIRECTIVE: "YML003",
+  BAD_DQ_ESCAPE: "YML004",
+  BAD_INDENT: "YML005",
+  BAD_PROP_ORDER: "YML006",
+  BAD_SCALAR_START: "YML007",
+  BLOCK_AS_IMPLICIT_KEY: "YML008",
+  BLOCK_IN_FLOW: "YML009",
+  DUPLICATE_KEY: "YML010",
+  IMPOSSIBLE: "YML011",
+  KEY_OVER_1024_CHARS: "YML012",
+  MISSING_ANCHOR: "YML013", // not available in TypeScript (reason for `| string`)
+  MISSING_CHAR: "YML014",
+  MULTILINE_IMPLICIT_KEY: "YML015",
+  MULTIPLE_ANCHORS: "YML016",
+  MULTIPLE_DOCS: "YML017",
+  MULTIPLE_TAGS: "YML018",
+  TAB_AS_INDENT: "YML019",
+  TAG_RESOLVE_FAILED: "YML020",
+  UNEXPECTED_TOKEN: "YML021",
+};
+
+export const YAML_RULES: ValidationRule[] = [
```

Comment:
@andreiv1992  Should you be able to cross reference the Monaco error with our internal ViolationResult than you can use `getRuleForResult(response, result)` and you get one of these descriptions that can give you additional metadata. (result here is the full lastResponse of the validator).